src/test/java/tests/Task40.java

```diff
@@ -0,0 +1,185 @@
+        driver.get("https://passport.yandex.com/auth/add?retpath=https%3A%2F%2Fpassport.yandex.com%2Fprofile&noreturn=1");
+
+        driver.findElement(By.id("passp-field-login")).sendKeys(name);
+        driver.findElement(By.cssSelector(".Button2.Button2_size_l.Button2_view_action")).click();
+        Thread.sleep(1500);//(ExplicitlyWait)
+        driver.findElement(By.xpath("//input[@dir = 'ltr']")).sendKeys(psw);
+        driver.findElement(By.cssSelector(".Button2.Button2_size_l.Button2_view_action")).click();
+        Thread.sleep(2500);//(ExplicitlyWait)
+
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector(".personal-info-login__text")));
+    }
+
+    @Test
+    void alertBoxTest() {
+        driver.get("https://demo.seleniumeasy.com/javascript-alert-box-demo.html");
+
+        driver.findElement(By.xpath("//button[@onclick  ='myAlertFunction()']")).click();
+        Alert alert = driver.switchTo().alert();
+        System.out.println(alert.getText());
+        alert.accept();
+    }
+
+    @Test
+    void alertConfirmBoxAccept() {
+        driver.get("https://demo.seleniumeasy.com/javascript-alert-box-demo.html");
+
+        driver.findElement(By.xpath("//button[@onclick  ='myConfirmFunction()']")).click();
+        Alert alert = driver.switchTo().alert();
+        System.out.println(alert.getText());
+        alert.accept();
+
+        WebElement result = wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("confirm-demo")));
+        Assertions.assertEquals(result.getText(), "You pressed OK!");
+    }
+
+    @Test
+    void alertConfirmBoxDismiss() {
+        driver.get("https://demo.seleniumeasy.com/javascript-alert-box-demo.html");
+
+        driver.findElement(By.xpath("//button[@onclick  ='myConfirmFunction()']")).click();
+        Alert alert = driver.switchTo().alert();
+        System.out.println(alert.getText());
+        alert.dismiss();
+
+        WebElement result = wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("confirm-demo")));
+        Assertions.assertEquals(result.getText(), "You pressed Cancel!");
+    }
+
+    @Test
+    void alertPromptBox() {
+        driver.get("https://demo.seleniumeasy.com/javascript-alert-box-demo.html");
+
+        driver.findElement(By.xpath("//button[@onclick  ='myPromptFunction()']")).click();
+        Alert alert = driver.switchTo().alert();
+        System.out.println(alert.getText());
+        alert.sendKeys("HELLO");
+        alert.accept();
+
+        WebElement result = wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("prompt-demo")));
+        Assertions.assertEquals(result.getText(), "You have entered 'HELLO' !");
+    }
+
+    @Test
+    void waitsForUser() {
+        driver.get("https://demo.seleniumeasy.com/dynamic-data-loading-demo.html");
+        driver.findElement(By.id("save")).click();
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//div[@id = 'loading']/img")));
```

Comment:
assert!. cause if you not assert, and just get wait Exception, that will be shown in report incorrectly